snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/async.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.IO;
+
+class Program
+{
+    static void Main(string[] args)
```

Comment:
Signature here would be `static async Task Main()`, and you'll need to add a `using` statement for hte System.Threading.Tasks namespace. In the text introducing the example, you should note that it requires C# 7.1 or later, where support for the async modifier on the entry point was added.

Reply:
Yes, it turns out I was using an older version of C#, so could not put the async on Main. Creating the separate method and putting the async modifier on that made it run. (Then I added named methods to all the other samples for symmetry, but I agree that was uncalled-for.)  
With the updated version of #C, I can put the async on Main, and if I add the System.Threading.Tasks, the code works. I do wonder if most people will have C# => 7.1, because 7.0 will be the default (latest major version).